src/routes/DigitalAssets/routes/Receive/DigitalAssetsReceiveViewContainer.js

```diff
@@ -3,18 +3,20 @@
 import { connect } from 'react-redux'
 
 import { selectWalletsAddressesList } from 'store/stateSelectors'
+import { selectActiveWallet } from 'store/selectors/wallets'
 import { reactRouterBack } from 'utils/browser'
+import { isVoid } from 'utils/type/'
 
 import DigitalAssetsReceiveView from './DigitalAssetsReceiveView'
 
-import {
-  openView,
-  closeView,
-} from './modules/digitalAssetsReceive'
-
 const mapStateToProps = (state: AppState) => {
-  const items: AddressNames = selectWalletsAddressesList(state)
-    .reduce((acc, address) => ({ ...acc, [address]: address }), {})
+  const walletsList: AddressNames = selectWalletsAddressesList(state)
+    .reduce((acc, address, index) => ({ ...acc, [address]: `Address #${index}` }), {})
+
+  const activeWallet = selectActiveWallet(state)
+  const walletAddress = activeWallet ? activeWallet.address : null
+  const items: AddressNames = isVoid(walletAddress) ?
+    walletsList : { [walletAddress]: 'Address #0' }
 
```

Comment:
on receive page we need to know several things:
1) current wallet
2) walletAddresses
3) addressNames map (to get names for addresses that was derived from mnemonic wallet)

`mapStateToProps` here will be very similar [to this one](https://github.com/jibrelnetwork/jwallet-web/blob/develop/src/layouts/MenuLayout/MenuLayoutContainer.js#L26-L43) [or another one](https://github.com/jibrelnetwork/jwallet-web/blob/develop/src/routes/Wallets/routes/Addresses/WalletsAddressesViewContainer.js#L24-L49)

I assume it's part of component's UI to decide which case to render (depends on `wallet.type`) - just one address for `address` wallet or selector with addresses (with names if they exist) for `mnemonic` wallet.

please check examples of usage of derived addresses from mnemonic

Reply:
Looks quite similar to me. Can you be more specific? What is wrong?